lib/src/executables/boilerplate_upgrade.dart

```diff
@@ -76,20 +94,28 @@ void main(List<String> args) {
   final shouldTreatAllComponentsAsPrivate =
       parsedArgs[_treatAllComponentsAsPrivateFlag] == true;
 
-  final query = FileQuery.dir(
-    pathFilter: (path) {
-      return isDartFile(path) && !isGeneratedDartFile(path);
-    },
-    recursive: true,
-  );
+  final pathArgs = parsedArgs.rest.isNotEmpty ? parsedArgs.rest : null;
+  // If a path or multiple paths are provided as a basic arg, get file paths for them.
+  Iterable<String> filePaths = [];
+  if (pathArgs != null) {
+    Set<String> pathSet = {};
+    pathArgs.forEach((path) =>
+        pathSet.addAll(filePathsFromGlob(Glob(path, recursive: true))));
+    filePaths = pathSet;
```

Comment:
You shouldn't need to create and resolve `Glob`s from the input paths - shell expansion happens before the executable is run, so something like this:

```
pub global run over_react_codemod:boilerplate_upgrade lib/**.dart
```

would result in `parsedArgs.rest` being a list of all the file paths within `lib/`, not the original `lib/**.dart` pattern.

The only downside with this approach that I'm realizing now is that `runInteractiveCodemod()` expects file paths, so if someone were to pass in a directory like `lib/` as a path, that would result in an error. We should probably have `runInteractiveCodemod` expand directory paths to include all of the files within that directory

Reply:
I take that last part back. If `runInteractiveCodemod` does its own expanding of directories, then we're right back where we were before with not really knowing which files to include by default.. e.g. if someone passes in `lib/` and we expand that, there might be SASS files or images in that directory that make no sense to run the codemod on. Leaving it up to the consumer to use an appropriate glob like `lib/**.dart` seems sufficient to me in this case.

Reply:
Addressed!